x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/hooks/use_latest_vulnerabilities_by_resource.ts

```diff
@@ -0,0 +1,240 @@
+import { showErrorToast } from '../../../common/utils/show_error_toast';
+import { FindingsBaseEsQuery } from '../../../common/types';
+
+type LatestFindingsRequest = IKibanaSearchRequest<SearchRequest>;
+type LatestFindingsResponse = IKibanaSearchResponse<SearchResponse<any, VulnerabilitiesAggs>>;
+
+interface VulnerabilitiesAggs {
+  count: AggregationsMultiBucketAggregateBase<AggregationsStringRareTermsBucketKeys>;
+  total: AggregationsCardinalityAggregate;
+  resources: AggregationsMultiBucketAggregateBase<FindingsAggBucket>;
+}
+
+interface FindingsAggBucket extends AggregationsStringRareTermsBucketKeys {
+  name: AggregationsMultiBucketAggregateBase<AggregationsStringTermsBucketKeys>;
+  region: AggregationsMultiBucketAggregateBase<AggregationsStringTermsBucketKeys>;
+  vulnerability_severity_critical: AggregationsSingleBucketAggregateBase;
+  vulnerability_severity_high: AggregationsSingleBucketAggregateBase;
+  vulnerability_severity_medium: AggregationsSingleBucketAggregateBase;
+  vulnerability_severity_low: AggregationsSingleBucketAggregateBase;
+}
+
+interface VulnerabilitiesQuery extends FindingsBaseEsQuery {
+  sortOrder: SortOrder;
+  enabled: boolean;
+  pageIndex: number;
+  pageSize: number;
+}
+
+export const getQuery = ({
+  query,
+  sortOrder = 'desc',
+  pageIndex,
+  pageSize,
+}: VulnerabilitiesQuery) => ({
+  index: LATEST_VULNERABILITIES_INDEX_PATTERN,
+  query: getSafeVulnerabilitiesQueryFilter(query),
+  aggs: {
+    total: { cardinality: { field: 'resource.id' } },
+    resources: {
+      terms: {
+        field: 'resource.id',
+        size: MAX_FINDINGS_TO_LOAD * 3,
+        // in case there are more resources then size, ensuring resources with more vulnerabilities
+        // will be included first, and then vulnerabilities with critical and high severity
+        order: [
+          {
+            _count: sortOrder,
+          },
+          {
+            vulnerability_severity_critical: 'desc',
+          },
+          {
+            vulnerability_severity_high: 'desc',
+          },
+          {
+            vulnerability_severity_medium: 'desc',
+          },
+        ],
+      },
+      aggs: {
+        vulnerabilitiesCountBucketSort: {
+          bucket_sort: {
+            sort: [
+              {
+                _count: { order: sortOrder },
+              },
+            ],
+            from: pageIndex * pageSize,
+            size: pageSize,
+          },
+        },
+        name: {
+          terms: { field: 'resource.name', size: 1 },
+        },
+        region: {
+          terms: { field: 'cloud.region', size: 1 },
+        },
+        vulnerability_severity_critical: {
+          filter: {
+            bool: {
+              filter: [
+                {
+                  bool: {
+                    should: [
+                      {
+                        term: {
+                          'vulnerability.severity': { value: VULNERABILITIES_SEVERITY.CRITICAL },
```

Comment:
Thanks, that worked just fine, I got my query out of a Dashboard query so it was probably more verbose than it needed to be.

Also in favor of consistency, I removed the `vulnerability_severity_` prefix, so that less verbose as well